### DISCUSSION TITLE
How to use the draw commands via the CLI?

### DISCUSSION THREAD

**Discussion by kuba-orlik** [libvips/libvips#2687] [Q&A] [answered]
Thanks for the awesome and super-fast lib!

I can't find anywhere instructions on how to use the draw commands with the CLI. I'm using:

```
vips VipsDrawRect canvas.png  1024 50 50 50 50
```

but that just leaves the `canvas.png` file untouched. Also, it's not obvious how to specify the color in the `ink` argument - the docs just say `value to draw`, and I can't figure out what format is expected.

**Comment by kuba-orlik** [comment, marked as answer]
Ok, I now figured out that I have to use a intermediary file:

```sh
vips copy cropped.png k2.v
vips draw_rect k2.v "255 0 0" 0 0 50 50 
vips copy k2.v with_rectangle.png
```

The part `"255 0 0"` tells vips to use a red color

**Reply by jcupitt** [reply]
Yes, the draw command directly modify the file, so the file must be a simple memory array held on disc. You can use things like binary PPM as well.
